src/operators/insert.ts

```diff
@@ -26,20 +27,30 @@ export class InsertOperator implements Operator {
   }
 
   static specification = {
+    defaultValue: { required: false, type: ['boolean,string,number,object'] },
     index: { required: false, type: ['number'] },
     select: { required: false, type: ['string'] },
     value: { required: false, type: ['boolean,string,number,object'] },
     position: { required: false, type: ['number'] },
   };
 
   handleData(data: any[]): any[] | null {
-    const { select: selection, value } = this.options;
+    const { defaultValue, select: selection, value } = this.options;
 
     if (selection && value !== undefined) {
-      throw new Error('function operator has both \'select\' and \'value\' options set');
+      throw new Error('insert operator has both \'select\' and \'value\' options set');
     }
 
-    const insertedValue = value || select(selection!, data[this.index]);
+    let insertedValue = value || select(selection!, data[this.index]);
+
+    if (insertedValue === undefined && defaultValue) {
```

Comment:
If `defaultValue` is a boolean false then wouldn't this fail to assign it? Perhaps this needs to check whether it's undefined?

Reply:
Yes, good catch.